climada/entity/exposures/litpop/litpop.py

```diff
@@ -0,0 +1,1199 @@
+                                                     [polygon], nodata=np.nan)
+    meta_out['total_population'] = total_population
+
+    lon, lat = u_coord.raster_to_meshgrid(meta_out['transform'],
+                                          meta_out['width'],
+                                          meta_out['height'])
+    gdf = geopandas.GeoDataFrame({'value': litpop_array.flatten()}, crs=meta_out['crs'],
+                                 geometry=geopandas.points_from_xy(lon.flatten(),
+                                                                   lat.flatten()))
+    gdf['latitude'] = lat.flatten()
+    gdf['longitude'] = lon.flatten()
+    if region_id is not None:
+        gdf['region_id'] = region_id
+    else:
+        gdf['region_id'] = u_coord.get_country_code(gdf.latitude, gdf.longitude)
+    # remove entries outside polygon and return:
+    return gdf.dropna(), meta_out
+
+def get_value_unit(fin_mode):
+    """get `value_unit` depending on `fin_mode`
+
+    Parameters
+    ----------
+    fin_mode : Socio-economic value to be used as an asset base
+
+    Returns
+    -------
+    value_unit : str
+
+    """
+    if fin_mode in ['none', 'norm']:
+        return ''
+    if fin_mode == 'pop':
+        return 'people'
+    return 'USD'
+
+def get_total_value_per_country(cntry_iso3a, fin_mode, reference_year, total_population=None):
+    """
+    Get total value for disaggregation, e.g., total asset value or population
+    for a country, depending on unser choice (fin_mode).
+
+    Parameters
+    ----------
+    cntry_iso3a : str
+        country iso3 code alphabetic, e.g. 'JPN' for Japan
+    fin_mode : str
+        Socio-economic value to be used as an asset base that is disaggregated
+        to the grid points within the country
+        * 'pc': produced capital (Source: World Bank), incl. manufactured or
+                built assets such as machinery, equipment, and physical structures
+                (pc is in constant 2014 USD)
+        * 'pc_land': produced capital (Source: World Bank), incl. manufactured or
+                built assets such as machinery, equipment, physical structures,
+                and land value for built-up land.
+                (pc is in constant 2014 USD)
+        * 'pop': population count (source: GPW, same as gridded population)
+        * 'gdp': gross-domestic product (Source: World Bank)
+        * 'income_group': gdp multiplied by country's income group+1
+        * 'nfw': non-financial wealth (Source: Credit Suisse, of households only)
+        * 'tw': total wealth (Source: Credit Suisse, of households only)
+        * 'norm': normalized by country
+        * 'none': LitPop per pixel is returned unscaled
+        The default is 'pc'
+    reference_year : int
+        reference year for data extraction
+    total_population : number, optional
+        total population number, only required for fin_mode 'pop'.
+        The default is None.
```

Comment:
The function does not access the whole input data (e.g. GPW), the confusion is probably because it is an internal function. Therefore, I will rename `get_total_value_per_country` to `_get_total_value_per_country`

Reply:
Be careful, hiding unclear functionalities behind private functions is not a good idea. This makes the code harder to maintain.

Reply:
understood, I have removed the in parameter `population` and changed the calling method to deal with the special case of `fin_mode=='pop'` outside the function (c.f. https://github.com/CLIMADA-project/climada_python/pull/267/commits/56088a8be6a8182df7a9872032eec16df2a9eb77 )